src/main/java/seedu/moneygowhere/logic/commands/ImportCommand.java

```diff
@@ -125,22 +135,41 @@ public CommandResult execute(Model model) throws CommandException {
         MappingIterator<Map<String, String>> it = mapper.readerFor(Map.class)
                 .with(schema)
                 .readValues(csvFile);
+
+        if (!it.hasNext()) {
+            throw new IOException(MESSAGE_NO_CONTENT);
+        }
         while (it.hasNext()) {
             count++;
             Map<String, String> rowAsMap = it.next();
-            if (rowAsMap.size() == 5) {
-                try {
-                    Spending spending = createSpending(rowAsMap);
-                    spendings.add(spending);
-                } catch (ParseException p) {
-                    errors.add("Row " + count + ": " + p.getMessage() + "\n");
-                }
+
+            if (!isValidCsv(rowAsMap)) {
+                throw new IOException(MESSAGE_INVALID_CSV);
+            }
+            try {
+                Spending spending = createSpending(rowAsMap);
+                spendings.add(spending);
+            } catch (ParseException p) {
+                errors.add("Row " + count + ": " + p.getMessage() + "\n");
             }
         }
 
         return new Pair<>(spendings, errors);
     }
 
+    /**
+     * Returns true if csv has the correct size and headers
+     * @param map A map representing a row of values from the csv file
+     */
+    public boolean isValidCsv(Map<String, String> map) {
+        if (map.size() != 5) {
+            return false;
+        } else {
+            String[] keys = "name|cost|date|remark|tagged".split("\\|");
```

Comment:
This a good approach. You can further improve the approach by moving the String array as a class level constant so that fields can be easily edited when the format changes.

e.g.
`String[] keys = { FIELD_NAME, FIELD_COST, FIELD_DATE ....`